projects/server/controller/productController.js

```diff
@@ -10,7 +10,12 @@ module.exports = {
 
 
             // run query
-            let products = await sequelize.query(`SELECT * FROM product ORDER BY RAND()`)
+            let products = await sequelize.query(`SELECT a.*
+            , CAST(COALESCE(sum(b.stock),0) AS UNSIGNED) total_stock 
+            FROM product a 
+            LEFT JOIN product_stock b
+            ON a.id = b.product_id
+            GROUP BY 1,2,3,4,5,6,7 ORDER BY RAND()`)
```

Comment:
it would be nice if sorted from ready stock products first

Reply:
updated